cpp/src/text/subword/data_normalizer.cu

```diff
@@ -0,0 +1,459 @@
+ */
+const codepoint_metadata_type* get_codepoint_metadata(cudaStream_t stream)
+{
+  return g_codepoint_metadata.find_or_initialize([stream](void) {
+    codepoint_metadata_type* table =
+      static_cast<codepoint_metadata_type*>(rmm::mr::get_default_resource()->allocate(
+        codepoint_metadata_size * sizeof(codepoint_metadata_type), stream));
+    thrust::fill(rmm::exec_policy(stream)->on(stream),
+                 table + cp_section1_end,
+                 table + codepoint_metadata_size,
+                 codepoint_metadata_default_value);
+    CUDA_TRY(cudaMemcpyAsync(table,
+                             codepoint_metadata,
+                             cp_section1_end * sizeof(codepoint_metadata[0]),  // 1st section
+                             cudaMemcpyHostToDevice,
+                             stream));
+    CUDA_TRY(cudaMemcpyAsync(
+      table + cp_section2_begin,
+      cp_metadata_917505_917999,
+      (cp_section2_end - cp_section2_begin + 1) * sizeof(codepoint_metadata[0]),  // 2nd section
+      cudaMemcpyHostToDevice,
+      stream));
+    return table;
+  });
+}
+
+/**
+ * @brief Retrieve the aux code point data table.
+ *
+ * Build the code point metadata tables in device memory
+ * using the vector pieces from codepoint_metadata.ah
+ */
+const aux_codepoint_data_type* get_aux_codepoint_data(cudaStream_t stream)
+{
+  return g_aux_codepoint_data.find_or_initialize([stream](void) {
+    aux_codepoint_data_type* table =
+      static_cast<aux_codepoint_data_type*>(rmm::mr::get_default_resource()->allocate(
+        aux_codepoint_data_size * sizeof(aux_codepoint_data_type), stream));
+    thrust::fill(rmm::exec_policy(stream)->on(stream),
+                 table + aux_section1_end,
+                 table + aux_codepoint_data_size,
+                 aux_codepoint_default_value);
+    CUDA_TRY(cudaMemcpyAsync(table,
+                             aux_codepoint_data,
+                             aux_section1_end * sizeof(aux_codepoint_data[0]),  // 1st section
+                             cudaMemcpyHostToDevice,
+                             stream));
+    CUDA_TRY(cudaMemcpyAsync(
+      table + aux_section2_begin,
+      aux_cp_data_44032_55203,
+      (aux_section2_end - aux_section2_begin + 1) * sizeof(aux_codepoint_data[0]),  // 2nd section
+      cudaMemcpyHostToDevice,
+      stream));
+    CUDA_TRY(cudaMemcpyAsync(
+      table + aux_section3_begin,
+      aux_cp_data_70475_71099,
+      (aux_section3_end - aux_section3_begin + 1) * sizeof(aux_codepoint_data[0]),  // 3rd section
+      cudaMemcpyHostToDevice,
+      stream));
+    CUDA_TRY(cudaMemcpyAsync(
+      table + aux_section4_begin,
+      aux_cp_data_119134_119232,
+      (aux_section4_end - aux_section4_begin + 1) * sizeof(aux_codepoint_data[0]),  // 4th section
+      cudaMemcpyHostToDevice,
+      stream));
+    return table;
+  });
+}
+
+}  // namespace
+
+data_normalizer::data_normalizer(uint32_t max_num_strings,
```

Comment:
These are the types from the original code so I did not want to change them since they may be based on other components up stream or down stream from this function.